cloudinit/sources/DataSourceAzure.py

```diff
@@ -1324,42 +1324,33 @@ def parse_network_config(imds_metadata):
                 nicname = 'eth{idx}'.format(idx=idx)
                 dev_config = {'dhcp4': False, 'dhcp6': False}
                 dhcp_override = {'route-metric': (idx + 1) * 100}
-                for addr4 in intf['ipv4']['ipAddress']:
-                    privateIpv4 = addr4['privateIpAddress']
-                    if privateIpv4:
-                        if dev_config.get('dhcp4', False):
-                            # Append static address config for ip > 1
-                            netPrefix = intf['ipv4']['subnet'][0].get(
-                                'prefix', '24')
-                            if not dev_config.get('addresses'):
-                                dev_config['addresses'] = []
-                            dev_config['addresses'].append(
-                                '{ip}/{prefix}'.format(
-                                    ip=privateIpv4, prefix=netPrefix))
-                        else:
+                for addr_type in ('ipv4', 'ipv6'):
+                    addresses = intf.get(addr_type, {}).get('ipAddress', [])
+                    if addresses:
+                        if addr_type == 'ipv4':
                             dev_config['dhcp4'] = True
                             # non-primary interfaces should have a higher
                             # route-metric (cost) so default routes prefer
                             # primary nic due to lower route-metric value
                             dev_config['dhcp4-overrides'] = dhcp_override
-                for addr6 in intf['ipv6']['ipAddress']:
-                    privateIpv6 = addr6['privateIpAddress']
-                    if privateIpv6:
-                        if dev_config.get('dhcp6', False):
-                            # Append static address config for ip > 1
-                            prefix_v6 = intf['ipv6']['subnet'][0].get(
-                                'prefix', '128')
-                            if not dev_config.get('addresses'):
-                                dev_config['addresses'] = []
-                            dev_config['addresses'].append(
-                                '{ip}/{prefix}'.format(
-                                    ip=privateIpv6, prefix=prefix_v6))
+                            default_prefix = '24'
                         else:
                             dev_config['dhcp6'] = True
                             # non-primary interfaces should have a higher
                             # route-metric (cost) so default routes prefer
                             # primary nic due to lower route-metric value
                             dev_config['dhcp6-overrides'] = dhcp_override
+                            default_prefix = '128'
```

Comment:
Let's add a comment around the slicing

The first IP is obtained via DHCP, we add all additional IPs as static addresses.